src/java/org/apache/cassandra/db/virtual/SystemViewsKeyspace.java

```diff
@@ -28,24 +30,28 @@
 
 public final class SystemViewsKeyspace extends VirtualKeyspace
 {
+    private static final boolean ONLY_LOCAL_AND_PEERS = Boolean.getBoolean("cassandra.system_view.only_local_and_peers_table");
+
     public static SystemViewsKeyspace instance = new SystemViewsKeyspace();
 
     private SystemViewsKeyspace()
     {
-        super(VIRTUAL_VIEWS, new ImmutableList.Builder<VirtualTable>()
-                    .add(new CachesTable(VIRTUAL_VIEWS))
-                    .add(new ClientsTable(VIRTUAL_VIEWS))
-                    .add(new SettingsTable(VIRTUAL_VIEWS))
-                    .add(new SystemPropertiesTable(VIRTUAL_VIEWS))
-                    .add(new SSTableTasksTable(VIRTUAL_VIEWS))
-                    .add(new ThreadPoolsTable(VIRTUAL_VIEWS))
-                    .add(new InternodeOutboundTable(VIRTUAL_VIEWS))
-                    .add(new InternodeInboundTable(VIRTUAL_VIEWS))
-                    .add(new SSTablesSystemView(VIRTUAL_VIEWS))
-                    .add(new SegmentsSystemView(VIRTUAL_VIEWS))
-                    .add(new IndexesSystemView(VIRTUAL_VIEWS))
-                    .add(new AnalyzerView(VIRTUAL_VIEWS))
-                    .addAll(TableMetricTables.getAll(VIRTUAL_VIEWS))
-                    .build());
+        super(VIRTUAL_VIEWS, ONLY_LOCAL_AND_PEERS ?
+                             Collections.EMPTY_LIST :
```

Comment:
If I understand the code correctly, it loads nothing when `system_view.only_local_and_peers_table` is given. However, the system property seems to require to load local system views, which is done in the [original patch](https://github.com/riptano/bdp/pull/19491/files#diff-a0664ac7179818d79a7c487b85f1ddb07d494e5e5c204fe2c45cec0a76ae07e5). Is it correct or are such views missing in CC?
It is good to mention that this is port of CNDB-1330.

Reply:
The correct views don't exist until STAR-1329 is merged. I'm happy for this one to wait until that is merged.